packages/calling-stateful-client/src/CallContext.ts

```diff
@@ -976,24 +976,34 @@ export class CallContext {
   }
   /* @conditional-compile-remove(close-captions) */
   private processNewCaption(captions: CaptionsInfo[], newCaption: CaptionsInfo): void {
-    // going through current captions to find the last caption said by the same speaker, remove that caption if it's partial and replace with the new caption
-    for (let index = captions.length - 1; index >= 0; index--) {
-      const currentCaption = captions[index];
+    // time stamp when new caption comes in
+    newCaption.timestamp = new Date();
```

Comment:
Do captions themselves have timestamp already?

Reply:
Their timestamp is when the audio gets first inputted to the captions bot and it doesn't get updated. So we need to use our own time stamp here